homeassistant/components/wled/coordinator.py

```diff
@@ -95,7 +96,12 @@ async def close_websocket(_: Event) -> None:
         )
 
         # Start listening
-        asyncio.create_task(listen())
+        def _done_callback(_: asyncio.Future) -> None:
+            """Cleanup hard task reference when future is done."""
+            self._listen_task = None
+
+        self._listen_task = asyncio.create_task(listen())
+        self._listen_task.add_done_callback(_done_callback)
```

Comment:
Instead of this, I would do:
```suggestion
        self.config_entry.async_on_unload(
            asyncio.create_task(listen()).cancel
        )
```

That way the listen task is automatically canceled when the entry is unloaded – because I don't see any way we're closing this connection now. Only the WLED device can cancel it?

Reply:
Although i guess that wouldn't clean up the STOP event listener.